File: UITests/UITests.m
//
//  UITests.m
//  UITests
//
//  Created by Sam Steele on 7/4/16.
//  Copyright © 2016 IRCCloud, Ltd. All rights reserved.
//

#import <XCTest/XCTest.h>
#import "UITests-Swift.h"

#define SCREENSHOT_DELAY 2

@interface UITests : XCTestCase

@end

@implementation UITests

- (void)setUp {
    [super setUp];
    self.continueAfterFailure = NO;
}

- (void)takeScreenshotTheme:(NSString *)theme mono:(BOOL)mono {
    [XCUIDevice sharedDevice].orientation = UIDeviceOrientationPortrait;
    
    XCUIApplication *app = [[XCUIApplication alloc] init];
    if (mono) {
        app.launchArguments = @[@"-theme", theme];
    } else {
        app.launchArguments = @[@"-theme", theme, @"-mono"];
    }
    [Snapshot setupSnapshot:app];
    [app launch];
    
    // XCUIElement *window = [app.windows elementBoundByIndex:0];
    
    // if (
    //     [theme isEqual: @"dawn"] ||
    //     window.horizontalSizeClass == UIUserInterfaceSizeClassCompact
    // ) {
        [NSThread sleepForTimeInterval:SCREENSHOT_DELAY];
        [Snapshot snapshot:[NSString stringWithFormat:@"%@-Portrait", theme]
                  waitForLoadingIndicator:NO];
    // }
    [XCUIDevice sharedDevice].orientation = UIDeviceOrientationLandscapeLeft;
    // if (
    //     window.horizontalSizeClass == UIUserInterfaceSizeClassRegular && (
    //         [theme isEqual: @"dawn"] ||
    //         window.verticalSizeClass == UIUserInterfaceSizeClassRegular
    //     )
    // ) {
        [NSThread sleepForTimeInterval:SCREENSHOT_DELAY];
        [Snapshot snapshot:[NSString stringWithFormat:@"%@-Landscape", theme]
                  waitForLoadingIndicator:NO];
    // }
}

- (void)testAshScreenshots {
    [self takeScreenshotTheme:@"ash" mono:YES];
}

- (void)testDawnScreenshots {
    [self takeScreenshotTheme:@"dawn" mono:NO];
}

- (void)testDuskScreenshots {
    [self takeScreenshotTheme:@"dusk" mono:NO];
}
@end

File: UITests/UITests.swift
//
//  UITests.swift
//  UITests
//
//  Copyright © 2016 IRCCloud, Ltd. All rights reserved.
//

import Foundation
import XCTest

let SCREENSHOT_DELAY:UInt32 = 2;

class UITests: XCTestCase {


override func setUp() {
    super.setUp()
    XCUIDevice().orientation = UIDeviceOrientation.portrait
}

  func takeScreenshotTheme(_ theme: String, mono: Bool = false, memberlist: Bool = false) {
    let app = XCUIApplication()
    app.launchArguments = ["-theme", theme]
    if (mono) {
        app.launchArguments += ["-mono"]
    }
    if (memberlist) {
        app.launchArguments += ["-memberlist"]
    }
    setupSnapshot(app)
    app.launch()
    
    let isPhone = UIDevice().userInterfaceIdiom == .phone
    let isPad = UIDevice().userInterfaceIdiom == .pad
    let args = app.launchArguments
    var isBigPhone = false
    if (
        args[args.endIndex-2] == "-FASTLANE_SNAPSHOT_DEVICE" &&
        args[args.endIndex-1] == "\"iPhone 6 Plus\""
    ) {
        isBigPhone = true
    }
    let isDawn = theme == "dawn"
    
    if (memberlist) {
        sleep(SCREENSHOT_DELAY)
        snapshot("\(theme)-Portrait-Members", waitForLoadingIndicator: false)
    } else {
        if (isDawn || isPhone) {
            sleep(SCREENSHOT_DELAY)
            snapshot("\(theme)-Portrait", waitForLoadingIndicator: false)
        }
        if (isPad || (isDawn && isBigPhone)) {
            XCUIDevice().orientation = UIDeviceOrientation.landscapeLeft;
            sleep(SCREENSHOT_DELAY)
            snapshot("\(theme)-Landscape", waitForLoadingIndicator: false)
        }
    }
}

func testAshScreenshots () {
    takeScreenshotTheme("ash", mono: true)
}
    
func testDawnScreenshots () {
    takeScreenshotTheme("dawn")
}
    
func testDawnMembersScreenshots () {
    if (UIDevice().userInterfaceIdiom == .phone) {
        takeScreenshotTheme("dawn", memberlist: true)
    }
}
    
func testDuskScreenshots () {
    takeScreenshotTheme("dusk")
}

}
